src/js_native_api_v8.cc

```diff
@@ -2984,3 +2984,26 @@ napi_status napi_adjust_external_memory(napi_env env,
 
   return napi_clear_last_error(env);
 }
+
+napi_status napi_set_instance_data(napi_env env,
```

Comment:
The semantics of calling this function multiple times are up for discussion. This is the simplest solution, and it's documented to not call the finalizer. Other options AFAICT are:

* Subsequent calls error.
* Subsequent calls replace the data/finalizer/hint and call the finalizer on the data already set.

I guess you're saying we should do the one that errors out. How about the other one?

Reply:
@mhdawson @addaleax WDYT? Any other options?

Reply:
If it errors out then there would be no way to clear the data.  One thought (not sure it's a good one) is that we could add a parameter  (say force) which must be true if you want to overwrite.  It would ensure that the developer specifically asked to overwrite, otherwise they'd get an error.

Reply:
@mhdawson @bnoordhuis @addaleax FWIW jerryscript also simply overwrites the pointer information without calling the finalizer and without requiring any overwrite flag.

Reply:
OTOH in the case of Node.js we have a greater chance that even a single addon may be controlled by multiple independent teams and they may need the ability to replace addon data in a controlled fashion, without knowing about each other's finalizers. Thus, calling the finalizer on a subsequent call may be the way to go 🤷

Reply:
I'm fine with just overwriting. It's been deliberated on now and it's properly documented. I'll mark this resolved.